File: example/example_package/example_package.cpp
#include "common_msgs.hpp"
#include <iostream>
#include <chrono>
#include <thread>

class Example_package {
public:
    void run() {
        while (true) {
            // TODO: Implement node logic
            std::cout << "Running example_package" << std::endl;
            std::this_thread::sleep_for(std::chrono::seconds(1));
        }
    }
};

int main(int argc, char** argv) {
    Example_package node;
    node.run();
    return 0;
}

File: example/example_package/example_package.go
package main

import (
    "fmt"
    "time"
    "common_msgs"
)

type Example_package struct{}

func (n *Example_package) Run() {
    for {
        // TODO: Implement node logic
        fmt.Println("Running example_package")
        time.Sleep(1 * time.Second)
    }
}

func main() {
    node := &Example_package{}
    node.Run()
}
